buildSrc/src/main/java/com/soundcloud/reflect/DaggerReflectPlugin.kt

```diff
@@ -1,20 +1,36 @@
 package com.soundcloud.reflect
 
+import org.gradle.api.GradleException
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.kotlin.dsl.dependencies
 import org.gradle.util.VersionNumber
 
 class DaggerReflectPlugin : Plugin<Project> {
     override fun apply(target: Project) {
+        if (target != target.rootProject) {
+            throw GradleException("Plugin must be applied to the root project")
+        }
+
         val extension = target.extensions.create("delect", DelectExtension::class.java)
 
-        if (!shouldActivateDaggerReflect(target)) {
-            return
+        target.afterEvaluate {
+            if (shouldActivateDaggerReflect(target, extension)) {
+                target.logger.warn("Using Dagger Reflect")
+                activateDaggerReflect(target, extension)
+            }
         }
+    }
 
-        target.logger.warn("Using Dagger Reflect")
+    private fun shouldActivateDaggerReflect(target: Project, extension: DelectExtension): Boolean =
```

Comment:
can we add a check to see if the gradle.property and the extension property are set to different values, we should fail the build?
or set an order of preference:
1. prefer the gradle property if it exists so we don't have to add the `afterEvaluate` block
2. check the extension

Reply:
The DSL value is only set after evaluation. If we do both checks in different places I think we overcomplicate things. I'd then rather vote to keep things as-is and don't introduce the DSL way at all.

After some local testing I figured that my clean builds actually became slower by utilizing the (original) plugin with partial reflection, so maybe kapt savings aren't really a thing for my current project anyways.

Reply:
The DSL value is only set after evaluation. If we do both checks in different places I think we overcomplicate things. I'd then rather vote to keep things as-is and don't introduce the DSL way at all.

After some local testing I figured that my clean builds actually became slower by utilizing the (original) plugin with partial reflection, so maybe kapt savings aren't really a thing for my current project anyways.

Reply:
Ok, nevermind me, my tests weren't good. I see a substantial speed up now.